arcjet/index.ts

```diff
@@ -310,11 +318,49 @@ export type EmailOptions = {
   allowDomainLiteral?: boolean;
 };
 
+type DetectEntities<T> = (
```

Comment:
I've renamed it for now. Is there any benefit to restricting the type further? I know that in Rust its a best practise to leave types as open as possible and then restrict further in the actual call sites (like how you can technically construct a HashMap<K, V> where V isn't Hash so its completely useless), but not sure if this is the same in TS.

Reply:
I think it's fine here because this type is not exported. If we exported it, we wouldn't want someone to be able to construct the type with a non-string because then it could never be applied to a function that works with the `detect` config option.